falcon/api.py

```diff
@@ -345,7 +345,8 @@ def add_route(self, uri_template, resource, *args, **kwargs):
         if '//' in uri_template:
             raise ValueError("uri_template may not contain '//'")
 
-        method_map = routing.create_http_method_map(resource)
+        method_map = routing.map_http_methods(resource)
+        method_map = routing.set_default_responders(method_map)
```

Comment:
Assigning `method_map` to the return value makes it unclear to the reader whether `set_default_responders()` mutates the passed-in `method_map` in place. In languages that do not allow side-effects like this, there is no question, but Python is a different matter. What do you think about not having the method return anything and removing the assignment?

```py
routing.set_default_responders(method_map)
```

Alternatively, this method could be renamed to `get_default_responders()` and it would not mutate the passed-in argument, but return a dict containing the extra responders that could then be merged with `method_map`. This would avoid the side-effect but I'm not certain it is worth it.